### ISSUE TITLE
[L] Add additional filters to A&R Export Report

### ISSUE THREAD

**Issue by kgorana** [bcgov/bcparks-ar-admin#265]
Description

As a park ranger I want to be able to get a report from A&R with additional filters showing things I am interested in so that I can quickly get that information without sifting through a giant spreadsheet.

This ticket is to add filtering on the A&R export report. Filters include: 
> [!IMPOARTNT]
> for refinement: add filters here (this has been iterated on during designs)

This will also prevent users from downloading a HUGE csv that either errors out or takes forever to download and navigate.

**Acceptance Criteria:**

* Export report page matches designs: [Attendance & Revenue (https://www.figma.com/file/yoMhvO0KZvsM6qoqwuIjpt/Attendance-%26-Revenue?type=design&node-id=3984%3A20917&mode=design&t=GOoQaKdGf95CBCd2-1) with new filters
  * Export contains filters for date, section, bundle, and park.
  * Park and sub-area list are based on role that user has: BRS-989: Enable Export A&R Report by Roles 
* Date range must be selected in order to run report (no max date selection required right now)
* If no filters are applied then it defaults to generate a report with all data available to that user role
* Exported csv report shows only what has been selected in filters

**Notes**

* most of the time parks staff (rec services) are looking back 3-5 years in the data and do this infrequently
* Original User Request from East Kootenay Ranger: Is there a way to filter down a report from the system? Seems like quite the excel spread sheet to search through when the entire province is downloaded.

**Linked issues**
is blocked by
- [ ] [BRS-993](https://bcparksdigital.atlassian.net/browse/BRS-993)
- [ ] [BRS-991](https://bcparksdigital.atlassian.net/browse/BRS-991)
- [ ] [BRS-989](https://bcparksdigital.atlassian.net/browse/BRS-989)
- [ ] [BRS-1141](https://bcparksdigital.atlassian.net/browse/BRS-1141)

**Designs**
https://www.figma.com/file/yoMhvO0KZvsM6qoqwuIjpt/Attendance-%26-Revenue?type=design&node-id=3984-20917&mode=design&t=HHrkBnoWYnkGCj3p-0

**Comment by kgorana** [comment]
**Jessica Wade** 
June 23, 2023 at 11:48 AM
@ndelraye I updated this ticket to include all filter stuff from the design work you’re doing. When ready can you attach designs. 
___
**Jessica Wade** 
April 12, 2023 at 4:25 PM
@ndelraye is there a way to add date range filters to this page as well?
In an effort to help with the issues we are seeing in [BRS-1115: Export reports in A&R prod not working - getting stuck on 63% ](https://bcparksdigital.atlassian.net/browse/BRS-1115) of too much data being there and user is unable to export. @cameronpettit thinking adding date filters to this page can be part of a solution.

**Comment by Dianadec** [comment]
This may have been duplicated here: https://github.com/bcgov/bcparks-ar-admin/issues/254

**Comment by LindsayMacfarlane** [comment]
October 11, 2024 prioritization session: A&R team identified this ticket/issue as something that can wait to be implemented (medium-priority). 

Mural board: https://app.mural.co/t/bcparks2575/m/bcparks2575/1728427370222/4c9714df914c7ec8bdcf20a9ee1999cc9e664a27?wid=0-1728427872927

**Comment by LindsayMacfarlane** [comment]
@davidclaveau - This A&R ticket has been flagged as a potential duplicate. Looking at the acceptance criteria, I believe everything has been implemented; however, can you take a look and confirm that is correct? Thanks :)

**Comment by davidclaveau** [comment]
I think it's mostly done and implemented. We have filtering by date now, and users will only receive an export with the parks associated with their roles.

I guess the only thing that is left is filtering by section, bundle, and park. It's not _overly_ _needed_ though as a user can filter in Excel by section, bundle, and park without issue, and right now we're not seeing an insane wait time for an Export without those filters (it's not like adding those filters would reduce the wait time from 20 minutes to 2 minutes - it's about a minute or two at most for a full export).

**Comment by LindsayMacfarlane** [comment]
That's sweet @davidclaveau. I'm going to close this ticket. If those filters are requested in the future we can always create a new ticket to capture.
